sqlite3_type.go

```diff
@@ -39,42 +60,47 @@ func (rc *SQLiteRows) ColumnTypeNullable(i int) (nullable, ok bool) {
 }
 
 // ColumnTypeScanType implement RowsColumnTypeScanType.
+// In SQLite3, this method should be called after Next() has been called, as sqlite3_column_type()
+// returns the column type for a specific row. If Next() has not been called, fallback to
+// sqlite3_column_decltype()
 func (rc *SQLiteRows) ColumnTypeScanType(i int) reflect.Type {
-	//ct := C.sqlite3_column_type(rc.s.s, C.int(i))  // Always returns 5
+	switch C.sqlite3_column_type(rc.s.s, C.int(i)) {
+	case C.SQLITE_INTEGER:
+		return TYPE_NULLINT
+	case C.SQLITE_FLOAT:
+		return TYPE_NULLFLOAT
+	case C.SQLITE_TEXT:
+		return TYPE_NULLSTRING
+	case C.SQLITE_BLOB:
+		return TYPE_RAWBYTES
+		//case C.SQLITE_NULL:
```

Comment:
Why is this commented out?

Reply:
Its to document that the omission is intentional and not a miss, this is the actual value that we would be getting in 2 cases:

1. If we call before we have a valid row
1. If the value for the row-column is NULL

Happy to remove it or add a comment clarifying.

Reply:
It looks like a mistake. An explicit comment is better.

Reply:
Added comment. Please take another look.

Reply:
👍